src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/UnclassifiedFast5Folder.groovy

```diff
@@ -34,9 +36,22 @@ class UnclassifiedFast5Folder extends DataFolder {
      * @return A new instance of a UnclassifiedFastQFolder object
      */
     static UnclassifiedFast5Folder create(String name, String relativePath, List<Fast5File> children) {
+        if (children) {
+            validateChildren(children)
+        }
         new UnclassifiedFast5Folder(name, relativePath, children)
     }
 
+    private static void validateChildren(List children) {
+        children.each { validateChild(it) }
+    }
+
+    private static void validateChild(Object child) {
+        if (!((child instanceof FastQFile) || (child instanceof FastQZippedFile))) {
```

Comment:
```suggestion
        if (!(child instanceof FastQFile) && !(child instanceof FastQZippedFile)) {
```
make it more declarative. This is only a suggestion choose whatever you like best.

Reply:
it was wrong anyway :)

Reply:
yes you needed to check for fast5 :D lol